compiler/p/codegen/PPCBinaryEncoding.cpp

```diff
@@ -1910,68 +2070,106 @@ void TR::PPCSrc3Instruction::fillBinaryEncodingFields(uint32_t *cursor)
       }
    }
 
+TR::RealRegister *toRealBaseRegister(TR::Instruction *instr, TR::Register *r)
+   {
+   return r ? toRealRegister(r) : instr->cg()->machine()->getRealRegister(TR::RealRegister::gr0);
+   }
+
+void fillMemoryReferenceD16RA(TR::Instruction *instr, uint32_t *cursor, TR::MemoryReference *mr)
+   {
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, !mr->getIndexRegister(), "Cannot use index-form MemoryReference with non-index-form instruction");
+   fillFieldD16(instr, cursor, mr->getOffset());
+   fillFieldRA(instr, cursor, toRealBaseRegister(instr, mr->getBaseRegister()));
+   }
+
+void fillMemoryReferenceDSRA(TR::Instruction *instr, uint32_t *cursor, TR::MemoryReference *mr)
+   {
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, !mr->getIndexRegister(), "Cannot use index-form MemoryReference with non-index-form instruction");
+   fillFieldD16(instr, cursor, mr->getOffset());
```

Comment:
Wrong fill routine!  fillFieldDS() is the right one.

Reply:
Nice catch! Fixed in dc1b1b616f48d6da8f205e7ee99d67f61e40ef40.